src/server/game/Entities/Object/Object.h

```diff
@@ -553,6 +553,7 @@ class WorldObject : public Object, public WorldLocation
 
     void DestroyForNearbyPlayers();
     virtual void UpdateObjectVisibility(bool forced = true, bool fromUpdate = false);
+	virtual void UpdateObjectVisibilityOnCreate() { UpdateObjectVisibility(true); }
```

Comment:
I don't see the utility of this method ? Why not just use UpdateObjectVisibility(true) ?

Reply:
I prefer to directly use UpdateObjectVisibility(true) and just add a description of why we do this. Although it's understandable, but technically you're adding extra instructions, work for the compiler, and lengthening the list of methods available on completion tools.

Reply:
Thanks for the feedback! I understand your point, and it makes sense to simplify by directly using UpdateObjectVisibility(true). I'll update the PR to remove the UpdateObjectVisibilityOnCreate method and include a descriptive comment explaining why this call is made. Let me know if there’s anything else you’d like me to adjust!

Reply:
I tried some fixes, but it generates some unwanted errors, especially in Modules. I believe this is ideal. If possible, you can do the Merge.